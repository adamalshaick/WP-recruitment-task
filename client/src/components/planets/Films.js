import React from "react";
import fetchDataDetails from "../hoc/fetchDataDetails";

export const Films = ({ data }) => {
  return <p>{data.title}</p>;
};

export default fetchDataDetails(Films);
