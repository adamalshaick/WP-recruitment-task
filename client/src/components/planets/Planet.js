import React, { Component } from "react";
import { TableRow, TableElement, TableName } from "../common/styles/Table";
import { PlanetWrapper } from "../common/styles/Planet";
import PlanetInfo from "./PlanetInfo";
import PropTypes from "prop-types";

export class Planet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showPlanetInfo: false
    };
  }

  onClick = e => {
    e.preventDefault();
    this.setState(prevState => ({
      showPlanetInfo: !prevState.showPlanetInfo
    }));
  };

  render() {
    return (
      <>
        <TableRow onClick={this.onClick} active={this.state.showPlanetInfo}>
          <TableName>{this.props.planet.name}</TableName>
          <TableElement>{this.props.planet.diameter}</TableElement>
          <TableElement>{this.props.planet.climate}</TableElement>
          <TableElement>{this.props.planet.gravity}</TableElement>
          <TableElement>{this.props.planet.population}</TableElement>
        </TableRow>
        <PlanetInfo
          active={this.state.showPlanetInfo}
          planet={this.props.planet}
        />
        <PlanetWrapper
          active={this.state.showPlanetInfo}
          onClick={this.onClick}
        />
      </>
    );
  }
}

Planet.propTypes = {
  planet: PropTypes.object.isRequired
};

export default Planet;
