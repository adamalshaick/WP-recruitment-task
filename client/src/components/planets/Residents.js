import React from "react";
import fetchDataDetails from "../hoc/fetchDataDetails";

export const Residents = ({ data }) => {
  return <p>{data.name}</p>;
};

export default fetchDataDetails(Residents);
