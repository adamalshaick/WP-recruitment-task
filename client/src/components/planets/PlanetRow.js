import React from "react";
import { PlanetCard, CardHeader } from "../common/styles/Planet";
import Films from "./Films";
import Residents from "./Residents";
import PropTypes from "prop-types";

export const PlanetRow = ({ active, planet }) => {
  return (
    <div>dd</div>
    // <PlanetCard active={active}>
    //   <CardHeader>
    //     {planet.name} planet details
    //     <hr />
    //   </CardHeader>
    //   <div style={{ display: "flex" }}>
    //     <div style={{ margin: "1rem" }}>
    //       <p>rotation period: {planet.rotation_period}</p>
    //       <p>orbital period: {planet.orbital_period}</p>
    //       <p>diameter: {planet.diameter}</p>
    //       <p>climate: {planet.climate}</p>

    //       <p>
    //         films:
    //         {active
    //           ? planet.films.map(film => {
    //               return <Films function={film} />;
    //             })
    //           : null}
    //       </p>
    //     </div>
    //     <div style={{ margin: "1rem" }}>
    //       <p>surface water: {planet.surface_water}</p>
    //       <p>population: {planet.population}</p>
    //       <p>terrain: {planet.terrain}</p>
    //       <p>gravity: {planet.gravity}</p>
    //       <p>
    //         residents:
    //         {active
    //           ? planet.residents.map(resident => {
    //               return <Residents function={resident} />;
    //             })
    //           : null}
    //       </p>
    //     </div>
    //   </div>
    // </PlanetCard>
  );
};

// PlanetRow.propTypes = {
//   planet: PropTypes.object.isRequired,
//   active: PropTypes.bool.isRequired
// };

export default PlanetRow;
