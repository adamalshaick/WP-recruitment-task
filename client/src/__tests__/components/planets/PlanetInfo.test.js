import React from "react";
import { PlanetInfo } from "../../../components/planets/PlanetInfo";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Films from "../../../components/planets/Films";
import Residents from "../../../components/planets/Residents";

describe("planet info functionality when active", () => {
  const Props = {
    planet: {
      films: [{ title: "title1" }, { title: "title2" }, { title: "title3" }],
      residents: [{ name: "resident1" }, { name: "resident2" }]
    },
    active: true
  };

  const wrapper = shallow(<PlanetInfo {...Props} />);
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("returns correct number of films component", () => {
    expect(wrapper.find(Films).length).toEqual(3);
  });

  it("returns correct number of residents component", () => {
    expect(wrapper.find(Residents).length).toEqual(2);
  });
});

describe("planet info functionality when disabled", () => {
  const Props = {
    planet: {
      films: [{ title: "title1" }, { title: "title2" }, { title: "title3" }],
      residents: [{ name: "resident1" }, { name: "resident2" }]
    },
    active: false
  };

  const wrapper = shallow(<PlanetInfo {...Props} />);
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("doenst render films and residents components", () => {
    expect(wrapper.find(Films).length).toEqual(0);
    expect(wrapper.find(Residents).length).toEqual(0);
  });
});
