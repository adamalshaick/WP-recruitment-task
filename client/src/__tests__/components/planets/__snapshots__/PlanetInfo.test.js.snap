// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`planet info functionality when active renders component 1`] = `
<styled.section
  active={true}
>
  <styled.header>
     planet details
    <hr />
  </styled.header>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "1rem",
        }
      }
    >
      <p>
        rotation period: 
      </p>
      <p>
        orbital period: 
      </p>
      <p>
        diameter: 
      </p>
      <p>
        climate: 
      </p>
      <p>
        films:
        <ComposedComponent
          function={
            Object {
              "title": "title1",
            }
          }
        />
        <ComposedComponent
          function={
            Object {
              "title": "title2",
            }
          }
        />
        <ComposedComponent
          function={
            Object {
              "title": "title3",
            }
          }
        />
      </p>
    </div>
    <div
      style={
        Object {
          "margin": "1rem",
        }
      }
    >
      <p>
        surface water: 
      </p>
      <p>
        population: 
      </p>
      <p>
        terrain: 
      </p>
      <p>
        gravity: 
      </p>
      <p>
        residents:
        <ComposedComponent
          function={
            Object {
              "name": "resident1",
            }
          }
        />
        <ComposedComponent
          function={
            Object {
              "name": "resident2",
            }
          }
        />
      </p>
    </div>
  </div>
</styled.section>
`;

exports[`planet info functionality when disabled renders component 1`] = `
<styled.section
  active={false}
>
  <styled.header>
     planet details
    <hr />
  </styled.header>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "1rem",
        }
      }
    >
      <p>
        rotation period: 
      </p>
      <p>
        orbital period: 
      </p>
      <p>
        diameter: 
      </p>
      <p>
        climate: 
      </p>
      <p>
        films:
      </p>
    </div>
    <div
      style={
        Object {
          "margin": "1rem",
        }
      }
    >
      <p>
        surface water: 
      </p>
      <p>
        population: 
      </p>
      <p>
        terrain: 
      </p>
      <p>
        gravity: 
      </p>
      <p>
        residents:
      </p>
    </div>
  </div>
</styled.section>
`;
